File: AuthenticationServices/src/main/java/com/artauction/authenticationservices/Entities/RefreshToken.java
package com.artauction.authenticationservices.Entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.time.LocalDateTime;

@Entity
@Table(name = "refresh_tokens")
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RefreshToken {
    // Attributes
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, unique = true)
    private String token;

    private LocalDateTime expiresOn;
    private LocalDateTime createdOn;
    private LocalDateTime revokedOn;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false)
    @OnDelete(action = OnDeleteAction.CASCADE)
    private User user;

    public boolean isActive() {
        return revokedOn == null && LocalDateTime.now().isBefore(expiresOn);
    }
}

File: Project/Backend/src/ArtAuction.Application/Entities/RefreshToken.cs
namespace ArtAuction.Application.Entities;

public class RefreshToken
{
    // Attributes
    public int Id { get; set; }
    public string Token { get; set; }
    public DateTime ExpiresOn { get; set; }
    public DateTime CreatedOn { get; set; }
    public DateTime? RevokedOn { get; set; }
    public bool IsActive => RevokedOn == null && DateTime.UtcNow < ExpiresOn;
    public int UserId { get; set; }
    
    // Relationships
    public ApplicationUser User { get; set; }
}